app/server/server.ts

```diff
@@ -156,6 +160,22 @@ Object.values(ProductTypes).forEach((productType: ProductType) => {
         })()
     )
   );
+
+  server.use(
```

Comment:
moved banner redirect into the dynamic routes (one per ProductType) rather than being a broad on any string